include/pybind11/detail/function_view.h

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2021 The Pybind Development Team.
+// All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+#pragma once
+
+#include "common.h"
+
+PYBIND11_NAMESPACE_BEGIN(PYBIND11_NAMESPACE)
+PYBIND11_NAMESPACE_BEGIN(detail)
+
+template <typename Signature>
+class function_view;
+
+template <typename R, typename... Arg>
```

Comment:
Would it make sense to explain how this compares to `std::function`?

Reply:
Done.